### DISCUSSION TITLE
Documentation & Demo; move to another repo?

### DISCUSSION THREAD

**Discussion by coclav** [maxGraph/maxGraph#20] [Development]
Hello team, 

I will setup the demos & documentation to be published online (jsgraph.github.io or something like that). Do you have a preference: 

- [ ] **maintain source in this repo**. Benefit is integration, one place for everything 
- [ ] **create a separate repo for demo / doc**. Benefit is that it can have its own lifecycle outside of the release of the product, different contributors and also makes the action pipeline cleaner as they just do one thing. 

I'm tempted for option 2 but wanted to have the insights of the community before ?

**Comment by coclav** [comment]
To clarify: i'm talking about "end user" documentation & demo and not the api documentation which is (in the current implementation) auto generated from the source code.

**Reply by junsikshim** [reply]
I'm currently working on reorganizing the @mcyph 's work into a monorepo and integrating Storybooks into it. That way, as we change the core code, we can easily see the changes it affected in the examples(both html/react).
I'll send a PR when it's somewhat ready so we can discuss about it :)

**Reply by tbouffard** [reply]
I am also in favor of keeping the user examples in the same repository using monorepo structure (as suggested by @junsikshim)  or whatever else.
IMHO, having examples in another repository introduces too much synchronization work/maintenance between the examples repository and the lib repository
- the lib used by examples must be updated when a new release is available
- when developing new features, extra stuffs must be done to create new examples using the new features as long as a new release of lib is not available publicly

**Reply by coclav** [reply]
ok great. Case closed. 

@junsikshim great idea to use storybook 👍
